app/server.go

```diff
@@ -291,20 +294,31 @@ func askReachableAddress(port string) network.Address {
 // rules etc).
 // In case anything goes wrong, an error is returned.
 func tryConnect(ip, binding network.Address) error {
-
 	stopCh := make(chan bool, 1)
+	listening := make(chan bool)
 	// let's bind
 	go func() {
 		ln, err := net.Listen("tcp", binding.NetworkAddress())
 		if err != nil {
 			log.Error("Trouble with binding to the address:", err)
 			return
 		}
+		listening <- true
 		con, _ := ln.Accept()
 		<-stopCh
 		con.Close()
 	}()
 	defer func() { stopCh <- true }()
+	<-listening
+	conn, err := net.Dial("tcp", ip.NetworkAddress())
+	if err != nil {
```

Comment:
That's wrong, sorry. Try to follow the error-path if err == nil and if err != nil. I proposed the following:

```
log.ErrFatal(err, "Could not connect itself to public address. This is most "+
			"probably an error in your system-setup. Please make sure this conode "+
			"can connect to", ip.NetworkAddress())

log.Info("Succesfully connected istelf to port")
conn.Close()
```